src/openms_gui/source/VISUAL/DIALOGS/ToolsDialog.cpp

```diff
@@ -400,6 +425,69 @@ namespace OpenMS
     }
   }
 
+  void ToolsDialog::reloadPlugins_()
+  {
+    const auto& tools = ToolHandler::getTOPPToolList();
+    const auto& utils = ToolHandler::getUtilList();
+    plugin_params_ = tool_scanner_->getPluginParams();
+
+    QStringList list;
+    for (auto& pair : tools)
+    {
+      std::vector<LayerDataBase::DataType> tool_types = getTypesFromParam_(tool_params_.copy(pair.first + ':'));
+      if (std::find(tool_types.begin(), tool_types.end(), layer_type_) != tool_types.end())
+      {
+        list << pair.first.toQString();
+      }
+    }
+    for (auto& pair : utils)
+    {
+      std::vector<LayerDataBase::DataType> tool_types = getTypesFromParam_(tool_params_.copy(pair.first + ":"));
+      if (std::find(tool_types.begin(), tool_types.end(), layer_type_) != tool_types.end())
+      {
+        list << pair.first.toQString();
+      }
+    }
+    //TODO: Plugins get added to the list just like tools/utils and can't be differentiated in the GUI
+    for (String name : tool_scanner_->getPlugins())
+    {
+      std::vector<LayerDataBase::DataType> tool_types = getTypesFromParam_(plugin_params_.copy(name + ":"));
+      if (std::find(tool_types.begin(), tool_types.end(), layer_type_) != tool_types.end())
+      {
+        list << name.toQString();
+      }
+    }
+
+    list.sort();
+    list.push_front("<select tool>");
+    int32_t selected_index = 0;
+    for (int32_t i = 0; i < list.size(); ++i)
+    {
+      if (list.at(i) == tools_combo_->currentText())
+      {
+        selected_index = i;
+        break;
+      }
+    }
+    if (selected_index == 0)
+    {
+      tool_desc_->clear();
+      arg_param_.clear();
+      vis_param_.clear();
+      editor_->clear();
+      input_combo_->clear();
+      output_combo_->clear();
+      disable_();
+    }
+    tools_combo_->clear();
+    tools_combo_->addItems(list);
+    if (selected_index > 0)
```

Comment:
this would be easier to understand if made part of `if .. else ` for the case above?